src/components/features/Registration/RegStepOne.vue

```diff
@@ -0,0 +1,175 @@
+<script>
+import { defineComponent } from 'vue';
+import AppInput from '@/components/common/App/AppInput.vue';
+import FormCard from '@/components/views/FormCard.vue';
+import ValidationMessage from '@/components/common/Form/ValidationMessage.vue';
+import AppButton from '@/components/common/App/AppButton.vue';
+import SectionTitle from '@/components/common/Section/SectionTitle.vue';
+import SectionSubtitle from '@/components/common/Section/SectionSubtitle.vue';
+
+export default defineComponent({
+  components: {
+    AppInput,
+    FormCard,
+    ValidationMessage,
+    AppButton,
+    SectionTitle,
+    SectionSubtitle,
+  },
+  data() {
+    return {
+      email: '',
+      password: '',
+      emailTouched: false,
+      passwordTouched: false,
+    }
+  },
+  emits: ['completed'],
+  computed: {
+    emailErrorMessage() {
+      if (!this.email && this.emailTouched) {
+        return 'Please enter your email'
+      }
+      if (this.email && this.emailTouched && !this.isEmailValid(this.email)) {
+        return 'Please enter CORRECT email'
+      }
+      return ''
+    },
+  },
+  methods: {
+    isEmailValid(input) {
+      return /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/.test(input)
+    },
+    hasOneDigit(input) {
+      return /\d/.test(input)
+    },
+    hasOneLetter(input) {
+      return /[a-zA-Z]/.test(input)
+    },
+    isLongEnough(input) {
+      return input.length >= 8
+    },
+    isPasswordValid(input) {
+      return this.password !== '' &&
+        this.passwordTouched === true &&
+        this.hasOneDigit(input) &&
+        this.hasOneLetter(input) &&
+        this.isLongEnough(input)
+    },
```

Comment:
In order to trim the code a little bit and have these functions testable, configurable (validating length but of varying size) and re-usable (I can imagine a form for updating user's password being a separate component which will want all these validations) we should try to extract all of these as helpers/place to some external place module and keep them there 👀